extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/WebSocketTelemetryRecorder.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.util.TypeLiteral;
+
+import io.quarkus.arc.SyntheticCreationalContext;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class WebSocketTelemetryRecorder {
+
+    public Function<SyntheticCreationalContext<WebSocketTelemetryProvider>, WebSocketTelemetryProvider> createTelemetryProvider() {
+        return new Function<>() {
+            @Override
+            public WebSocketTelemetryProvider apply(SyntheticCreationalContext<WebSocketTelemetryProvider> ctx) {
+                Instance<Consumer<WebSocketTelemetryProviderBuilder>> builderCustomizers = ctx
+                        .getInjectedReference(new TypeLiteral<>() {
```

Comment:
This should be `.getInjectedReference(new TypeLiteral<Instance<Consumer<WebSocketTelemetryProviderBuilder>>>() {}`. The `new TypeLiteral<>() {}` represents `Object` which works because all beans have the bean type `java.lang.Object` but it's better to be explicit here.

Reply:
I don't really understand why `new TypeLiteral<>() {}` would represent `Object`, it defy what I thought I knew about Java:

```
Instance<Consumer<WebSocketTelemetryProviderBuilder>> builderCustomizers = ctx
                        .getInjectedReference(new TypeLiteral<>() {});

<R> R getInjectedReference(TypeLiteral<R> requiredType, Annotation... qualifiers);  --> R is Consumer<WebSocketTelemetryProviderBuilder>
```

That said, I'll apply your suggestion.

Reply:
Ah, you're right, forget about my comment then. I forgot that `getInjectedReference()` is generic...